File: leetcode523.py
nums = [0]
k = 1

if k == 0:
    for i in range(len(nums) - 1):
        if nums[i] == 0 and nums[i + 1] == 0:
            print(True)
            break
    else:
        print(False)
else:
    remainderdict = {0: -1}
    cumulativesum = 0

    for i in range(len(nums)):
        cumulativesum += nums[i]
        remainder = cumulativesum % k

        if remainder in remainderdict:
            if i - remainderdict[remainder] > 1:
                print(True)
                break
        else:
            remainderdict[remainder] = i
    else:
        print(False)

File: leetcode523.c
#include <stdio.h>

int main() {
    int nums[] = {0};
    int k = 1;
    int numsSize = sizeof(nums) / sizeof(nums[0]);
    
    if (k == 0) {
        for (int i = 0; i < numsSize - 1; i++) {
            if (nums[i] == 0 && nums[i + 1] == 0) {
                printf("true\n");
                return 0;
            }
        }
        printf("false\n");
    } else {
        int remainderDict[numsSize + 1];
        int cumulativeSum = 0;

        for (int i = 0; i <= numsSize; i++) {
            remainderDict[i] = -1;
        }

        for (int i = 0; i < numsSize; i++) {
            cumulativeSum += nums[i];
            int remainder = cumulativeSum % k;

            if (remainderDict[remainder] != -1) {
                if (i - remainderDict[remainder] > 1) {
                    printf("true\n");
                    return 0;
                }
            } else {
                remainderDict[remainder] = i;
            }
        }
        printf("false\n");
    }

    return 0;
}
